nni/experiment/config/convert.py

```diff
@@ -1,330 +1,257 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-import json
 import logging
-from pathlib import Path
-from tempfile import NamedTemporaryFile
-from typing import Any, Dict, List
 
-from .common import ExperimentConfig
+from .common import ExperimentConfig, AlgorithmConfig, CustomAlgorithmConfig
+from .remote import RemoteMachineConfig
+from .kubeflow import KubeflowRoleConfig, KubeflowNfsConfig, KubeflowAzureStorageConfig
+from .frameworkcontroller import FrameworkControllerRoleConfig
+from .shared_storage import NfsConfig, AzureBlobConfig
 from . import util
 
 _logger = logging.getLogger(__name__)
 
-
-def to_v1_yaml(config: ExperimentConfig, skip_nnictl: bool = False) -> Dict[str, Any]:
-    config.validate(False)
-    data = config.json()
-
-    ts = data.pop('trainingService')
-
-    data['trial'] = {
-        'command': data.pop('trialCommand'),
-        'codeDir': data.pop('trialCodeDirectory'),
-    }
-
-    if 'trialGpuNumber' in data:
-        data['trial']['gpuNum'] = data.pop('trialGpuNumber')
-
-    if isinstance(ts, list):
-        hybrid_names = []
-        for conf in ts:
-            if conf['platform'] == 'openpai':
-                conf['platform'] = 'pai'
-            hybrid_names.append(conf['platform'])
-            _handle_training_service(conf, data)
-        data['trainingServicePlatform'] = 'hybrid'
-        data['hybridConfig'] = {'trainingServicePlatforms': hybrid_names}
-    else:
-        if ts['platform'] == 'openpai':
-            ts['platform'] = 'pai'
-        data['trainingServicePlatform'] = ts['platform']
-        _handle_training_service(ts, data)
-
-    data['authorName'] = 'N/A'
-    data['experimentName'] = data.get('experimentName', 'N/A')
-    data['maxExecDuration'] = data.pop('maxExperimentDuration', '999d')
-    if data['debug']:
-        data['versionCheck'] = False
-    data['maxTrialNum'] = data.pop('maxTrialNumber', 99999)
-
-    ss = data.pop('searchSpace', None)
-    ss_file = data.pop('searchSpaceFile', None)
-    if ss is not None:
-        ss_file = NamedTemporaryFile('w', delete=False)
-        json.dump(ss, ss_file, indent=4)
-        data['searchSpacePath'] = ss_file.name
-    elif ss_file is not None:
-        data['searchSpacePath'] = ss_file
-    if 'experimentWorkingDirectory' in data:
-        data['logDir'] = data.pop('experimentWorkingDirectory')
+def to_v2(v1) -> ExperimentConfig:
+    platform = v1.pop('trainingServicePlatform')
+    assert platform in ['local', 'remote', 'openpai']
```

Comment:
what if `aml` `kubeflow` ...

Reply:
K8S based will stick with v1.
AML is a mistake and has been added.